colibre/scripts/deadtime_evolution.py

```diff
@@ -0,0 +1,90 @@
+"""
+Plots dead time fraction v.s. step size
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from swiftpipeline.argumentparser import ScriptArgumentParser
+from glob import glob
+import re
+import scipy.stats as stats
+
+arguments = ScriptArgumentParser(
+    description="Creates a run performance plot: evolution of the dead time fraction in scale factor bins"
+)
+
+run_names = arguments.name_list
+run_directories = [f"{directory}" for directory in arguments.directory_list]
+output_path = arguments.output_directory
+
+plt.style.use(arguments.stylesheet_location)
+
+fig, ax = plt.subplots()
+
+color_index = 0
+for run_name, run_directory in zip(run_names, run_directories):
+
+    color = f"C{color_index}"
+    color_index += 1
+
+    timesteps_glob = glob(f"{run_directory}/timesteps_*.txt")
+    timesteps_filename = timesteps_glob[0]
+
+    # extract the number of ranks from the file
+    nrank = 1
+    nthread = 1
+    with open(timesteps_filename, "r") as file:
+        header = file.readlines()[:10]
+        for line in header:
+            match = re.findall("Number of MPI ranks: (\d+)", line)
+            if len(match) > 0:
+                nrank = int(match[0])
+            match = re.findall("Number of threads: (\d+)", line)
+            if len(match) > 0:
+                nthread = int(match[0])
+    try:
+        data = np.genfromtxt(
+            timesteps_filename,
+            skip_footer=5,
+            loose=True,
+            invalid_raise=True,
+            usecols=(2, 12, 14),
+            dtype=[
+                ("a", "f4"),
+                ("wallclock", "f4"),
+                ("deadtime", "f4"),
+            ],
+        )
+    except:
```

Comment:
Maybe it is worth being more specific and writing here which exceptions we expect to catch here? I guess? `FileNotFoundError` and `ValueError`?

Reply:
It is mostly meant to catch `ValueError`s because of old timestep files that do not have the deadtime column. I realised that also means you need to set `loose=False` in the function call, so I have updated that as well.